cdap-app-fabric/src/main/java/co/cask/cdap/metadata/MetadataHttpHandler.java

```diff
@@ -609,8 +611,7 @@ public void getProgramTags(HttpRequest request, HttpResponder responder,
                              @PathParam("program-type") String programType,
                              @PathParam("program-id") String programId,
                              @QueryParam("scope") String scope) throws NotFoundException, BadRequestException {
-    Id.Program program = Id.Program.from(Id.Application.from(namespaceId, appId),
-                                         ProgramType.valueOfCategoryName(programType), programId);
+    ProgramId program = new ProgramId(namespaceId, appId, ProgramType.valueOf(programType), programId);
```

Comment:
should use ``ProgramType.valueOfCategoryName``. How come this does not break anything? Can you check elsewhere too?